lib/PuppeteerSharp/Extensions.cs

```diff
@@ -62,6 +62,36 @@ public static async Task<T> EvaluateFunctionAsync<T>(this ElementHandle elementH
             return result;
         }
 
+        /// <summary>
+        /// Runs <paramref name="pageFunction"/> within the frame and passes it the outcome the <paramref name="elementHandle"/> as the first argument
+        /// </summary>
+        /// <typeparam name="T">The type of the response</typeparam>
+        /// <param name="elementHandle">An <see cref="ElementHandle"/> that will be used as the first argument in <paramref name="pageFunction"/></param>
+        /// <param name="pageFunction">Function to be evaluated in browser context</param>
+        /// <param name="disposeHandle">If set to <c>false</c> the <paramref name="elementHandle"/> will not be disposed. Only opt out of disposal if the <paramref name="pageFunction"/> is a pure function with no side effects on the <paramref name="elementHandle"/>.</param>
+        /// <param name="args">Arguments to pass to <c>pageFunction</c></param>
+        /// <returns>Task which resolves to the return value of <c>pageFunction</c></returns>
+        /// <exception cref="SelectorException">If <paramref name="elementHandle"/> is <c>null</c></exception>
+        public static async Task<T> EvaluateFunctionAsync<T>(this ElementHandle elementHandle, string pageFunction, bool disposeHandle, params object[] args)
```

Comment:
I'm not against this feature, but this would be a breaking change for people sending a `bool` as the first `params object[] args`

Reply:
I have 2 ideas here.
 1. A property on the ElementHandle. `AutoDispose`.
 2. Move the `disposeHandle`to the second position on this signature.

Thoughts @Meir017?

Reply:
Good catch